src/Web/Base/EntityModificationAction.php

```diff
@@ -53,6 +52,21 @@ protected function validateModelAndTryToSaveEntity() {
         return $result;
     }
 
+    protected function handleSuccessfulOperation() {
+        if ($this->redirectUrl) {
+            return $this->redirectToNextPage();
+        }
+        $model = $this->getModel();
+
+        return $this->renderViewFile(compact('model'));
+    }
+
+    protected function handledFailedOperation() {
+        $model = $this->getModel();
+
+        return $this->renderViewFile(compact('model'));
```

Comment:
I think using `compact` here is redundant as there is only one variable - model. We can have it as 
```
$this->renderViewFile([
    'model' => $this->getModel(),
]);
```

And the same goes for "success" method.